src/runtimes/node/bundlers/nft/transpile.ts

```diff
@@ -1,32 +1,23 @@
 import { build } from '@netlify/esbuild'
-import { tmpName } from 'tmp-promise'
 
 import type { FunctionConfig } from '../../../../config'
-import { safeUnlink } from '../../../../utils/fs'
 import { getBundlerTarget } from '../esbuild/bundler_target'
 
 const transpile = async (path: string, config: FunctionConfig) => {
-  const targetPath = await tmpName({ postfix: '.js' })
-  const cleanupFn = () => safeUnlink(targetPath)
-
   // The version of ECMAScript to use as the build target. This will determine
   // whether certain features are transpiled down or left untransformed.
   const nodeTarget = getBundlerTarget(config.nodeVersion)
-
-  await build({
+  const transpiled = await build({
```

Comment:
We were piping the output of esbuild to a temporary file, but actually we can just get the result loaded into memory by setting `write: false` and avoid creating a file altogether.